src/multi.c

```diff
@@ -35,39 +35,17 @@
 #include "redisraft.h"
 
 
-typedef struct MultiState {
-    RaftRedisCommandArray cmds;
-    bool error;
-} MultiState;
-
-void MultiInitClientState(RedisRaftCtx *rr)
-{
-    rr->multi_client_state = RedisModule_CreateDict(NULL);
-}
-
-uint64_t MultiClientStateCount(RedisRaftCtx *rr)
+void ResetMultiClientState(ClientState * clientState)
```

Comment:
Looks like multi.c is not necessary anymore (it's getting shorter). If you want, we can delete the file and move the function here back into the redisraft.c.

Reply:
So I was thinking sort of the opposite, it be nicer if different commands were handled in different files.

Reply:
In general, I find harder to read/navigate multiple files, especially on github.  (Also, it reminds me my Java days that you have billion of files even in the simplest project, [link](https://github.com/EnterpriseQualityCoding/FizzBuzzEnterpriseEdition)). 

If you have bunch of commands that you handle in 10 lines of code, creating a file for each seems overkill to me but anyway, I don't have a strong opinion.

Reply:
Agree with @sjpotter, I think we should separate logic into different files, as long as we can.